File: question/neetcode150/binary_search/binary-search.py
from typing import List

class Solution:
    def search(self, a: List[int], t: int) -> int:
        i = 0
        n = len(a)
        while n:
            while i+n < len(a) and a[i+n] < t:
                i+=n
            n//=2

        return -1 if a[i] != t else i

File: question/neetcode150/binary_search/binary-search.cpp
#include <bits/stdc++.h>

using namespace std;

class Solution {
public:
    int search(vector<int>& a, int t) {
        int n = a.size();
        int k = 0;

        for(int i = n; i > 0;i/=2){
            while(k+i<n and a[k+i]<=t){
                k+=i;
            }
        }

        if(a[k]==t){
            return k;
        }
        return -1;
    }
};